pilot/cmd/pilot-agent/main.go

```diff
@@ -476,13 +479,19 @@ func getDNSDomain(domain string) string {
 	return domain
 }
 
-// check if SDS UDS path and token path exist, if both exist, requests key/cert
-// using SDS instead of secret mount.
+// detectSds checks config for control plane SDS . The returned values are used by caller
+// to decide whether SDS or file-mounted secret is used for control plane.
 func detectSds(controlPlaneBootstrap, controlPlaneAuthEnabled bool, udspath, trustworthyJWTPath string) (bool, string) {
 	if !sdsEnabledVar.Get() {
 		return false, ""
 	}
 
+	// for controlplane sidecar, if controlplanesecurity isn't enabled
```

Comment:
Remove "for controlplane sidecar"?
My understanding is, the SDS is only used to set the communication from Envoy to Pilot. So, if controlPlaneAuthEnabled is false, SDS doesn't need to be set.

Reply:
It is not only about connection to Pilot, but also communication between control plane component sidecars. In #16459, all the SDS config changes and existing file-mount config under pilot/docker/ are protected by {{- if .ControlPlaneAuth }}, and this indicates all the affected communication between workload sidecar to Pilot sidecar, and between control plane sidecars.
changes for communication between workload sidecar and Pilot sidecar is here tools/packaging/common/envoy_bootstrap_v2.json in #16459

Reply:
I see. Thanks.